src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+                                            const unsigned int pointerUpdateKind,
+                                            const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                            const bool focused,
+                                            const til::point terminalPosition)
+    {
+        // Short-circuit isReadOnly check to avoid warning dialog
+        if (focused && !_core->IsInReadOnlyMode() && _canSendVTMouseInput(modifiers))
+        {
+            _trySendMouseEvent(pointerUpdateKind, buttonState, modifiers, terminalPosition);
+        }
+        else if (focused && buttonState.isLeftButtonDown)
+        {
+            if (_singleClickTouchdownPos)
+            {
+                // Figure out if the user's moved a quarter of a cell's smaller axis away from the clickdown point
+                auto& touchdownPoint{ *_singleClickTouchdownPos };
+                auto distance{ std::sqrtf(std::powf(mouseCursorPosition.X - touchdownPoint.X, 2) +
+                                          std::powf(mouseCursorPosition.Y - touchdownPoint.Y, 2)) };
+
+                const auto fontSizeInDips{ _core->FontSizeInDips() };
+                if (distance >= (std::min(fontSizeInDips.width(), fontSizeInDips.height()) / 4.f))
+                {
+                    _core->SetSelectionAnchor(terminalPosition);
+                    // stop tracking the touchdown point
+                    _singleClickTouchdownPos = std::nullopt;
+                    _singleClickTouchdownTerminalPos = std::nullopt;
+                }
+            }
+
+            SetEndSelectionPoint(terminalPosition);
+
+            // TODO: Make sure this still works
+            //
+            // const double cursorBelowBottomDist = cursorPosition.Y - SwapChainPanel().Margin().Top - SwapChainPanel().ActualHeight();
+            // const double cursorAboveTopDist = -1 * cursorPosition.Y + SwapChainPanel().Margin().Top;
+
+            // constexpr double MinAutoScrollDist = 2.0; // Arbitrary value
+            // double newAutoScrollVelocity = 0.0;
+            // if (cursorBelowBottomDist > MinAutoScrollDist)
+            // {
+            //     newAutoScrollVelocity = _getAutoScrollSpeed(cursorBelowBottomDist);
+            // }
+            // else if (cursorAboveTopDist > MinAutoScrollDist)
+            // {
+            //     newAutoScrollVelocity = -1.0 * _getAutoScrollSpeed(cursorAboveTopDist);
+            // }
+
+            // if (newAutoScrollVelocity != 0)
+            // {
+            //     _tryStartAutoScroll(point, newAutoScrollVelocity);
+            // }
+            // else
+            // {
+            //     _tryStopAutoScroll(ptr.PointerId());
+            // }
+        }
+
+        _core->UpdateHoveredCell(terminalPosition);
+    }
+
+    void ControlInteractivity::TouchMoved(const winrt::Windows::Foundation::Point newTouchPoint,
+                                          const bool focused)
+    {

```

Comment:
null-check `_core`?

Reply:
We instantiate `_core` in the ctor, so this isn't really needed